### ISSUE TITLE
Option to ignore non-final fields expected to be used for equals

### ISSUE THREAD

**Issue by abika** [jqno/equalsverifier#200]
Considering this simple example (almost auto-generated by IDE):
```
public class Foo {

    private final long id;
    private final String name;
    private String description = "";

    public Foo(long id, String name, String description) {
        this.id = id;
        this.name = name;
        this.description = description;
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Foo foo = (Foo) o;
        return id == foo.id && Objects.equals(name, foo.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name);
    }
}
```

`EqualsVerifier.forClass(Foo.class).usingGetClass().verify();` does fail with
> java.lang.AssertionError: Significant fields: equals does not use description, or it is stateless.

I follow the common practice to not use non-final fields for equals (insert foos into a set and later modify them -> ugh!).
Now there is the option to explicitly tell EqualsVerifier the fields used in equals -> not practical if EqualsVerifier should be used for 100+ already existing classes. Or `suppress(Warning.ALL_FIELDS_SHOULD_BE_USED)`, but this does not differentiate between final and non-final fields. 

So i kindly propose a new `suppress(Warning.ALL_NON_FINAL_FIELDS_SHOULD_BE_USED)`.

**Comment by jqno** [comment]
Hi Alex,

That sounds like a good idea.
I'm on holiday now so I'll get back to you about this later!

**Comment by jqno** [comment]
I just released version 2.5 with a `Warning.ALL_NONFINAL_FIELDS_SHOULD_BE_USED` flag.

**Comment by abika** [comment]
Thanks for the fast implementation!
